charts/nitro/templates/_helpers.tpl

```diff
@@ -149,6 +149,44 @@ Currently primarily used for stateless validator configuration
 {{- /* Return the processed JSON data */ -}}
 {{- $processed -}}
 
+{{- if and .Values.validator .Values.validator.splitvalidator .Values.validator.splitvalidator.global .Values.validator.splitvalidator.global.resources .Values.validator.splitvalidator.global.resources.limits .Values.validator.splitvalidator.global.resources.limits.memory -}}
+{{- $memory := .Values.validator.splitvalidator.global.resources.limits.memory -}}
+{{- $value := regexFind "^\\d*\\.?\\d+" $memory | float64 -}}
+{{- $unit := regexFind "[A-Za-z]+" $memory -}}
+{{- $valueMi := 0.0 -}}
+{{- if eq $unit "Gi" -}}
+  {{- $valueMi = mulf $value 1024 -}}
+{{- else if eq $unit "Mi" -}}
+  {{- $valueMi = $value -}}
+{{- end }}
+{{- if $.Values.env.goMemLimit.enabled }}
+- name: GOMEMLIMIT
+  value: {{ printf "%dMiB" (int (mulf $valueMi ($.Values.env.goMemLimit.multiplier | default 0.75))) }}
+{{- end }}
+{{- end -}}
+
 {{- end -}}
 
 
+{{- define "nitro.env" -}}
+{{- if and .Values.resources .Values.resources.limits .Values.resources.limits.memory -}}
+{{- $memory := .Values.resources.limits.memory -}}
+{{- $value := regexFind "^\\d*\\.?\\d+" $memory | float64 -}}
+{{- $unit := regexFind "[A-Za-z]+" $memory -}}
+{{- $valueMi := 0.0 -}}
+{{- if eq $unit "Gi" -}}
+  {{- $valueMi = mulf $value 1024 -}}
+{{- else if eq $unit "Mi" -}}
+  {{- $valueMi = $value -}}
+{{- end }}
+{{- if $.Values.env.goMemLimit.enabled }}
+- name: GOMEMLIMIT
+  value: {{ printf "%dMiB" (int (mulf $valueMi ($.Values.env.goMemLimit.multiplier | default 0.9))) }}
+{{- end }}
+{{- if $.Values.env.resourceMgmtMemFreeLimit.enabled }}
+- name: NITRO_NODE_RESOURCE__MGMT_MEM__FREE__LIMIT
```

Comment:
We are assuming the [env prefix](https://github.com/OffchainLabs/community-helm-charts/blob/0d8255eb12d9fc8e53110c79320ef1e854e00979/charts/nitro/values.yaml#L190) is `NITRO`. 

We can do that internally, but users may change that, so we should check that 
1. It's set (templating should fail if this feature is enabled and it is not set)
2. Directly use the value it is set to